spec/classes/plugin/nginx_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+
+describe 'letsencrypt::plugin::nginx' do
+  on_supported_os.each do |os, facts|
+    context "on #{os} based operating systems" do
+      let(:facts) { facts }
+      let(:params) { {} }
+      let(:pre_condition) do
+        <<-PUPPET
+        class { 'letsencrypt':
+          email => 'foo@example.com',
+        }
+        PUPPET
+      end
+
+      context 'with default parameters' do
+        it { is_expected.to compile.with_all_deps }
+
+        package_name =
+          if facts[:os]['family'] == 'RedHat' && facts[:os]['release']['major'] == '7'
+            'python2-certbot-nginx'
+          else
+            'python3-certbot-nginx'
+          end
```

Comment:
I think `let` is cleaner for these multi line statements.
```suggestion
        let(:package_name) do
          if facts[:os]['family'] == 'RedHat' && facts[:os]['release']['major'] == '7'
            'python2-certbot-nginx'
          else
            'python3-certbot-nginx'
          end
        end
```

Reply:
I have to say, I don't know much about rspec and why let is the cleaner way^^ If you don't mind pointing me in the right direction, so I can learn something. :)

Reply:
https://stackoverflow.com/questions/5359558/when-to-use-rspec-let/5359979#5359979 does a good job and actually what I myself used. https://relishapp.com/rspec/rspec-core/docs/helper-methods/let-and-let is the official documentation.

In this particular case I'd prefer it because it lets you easily write multi-line statements.